node/channel.js

```diff
@@ -606,6 +612,13 @@ TChannel.prototype.request = function channelRequest(options) {
     var self = this;
 
     options = options || {};
+
+    if (options.hasNoParent && (Math.random() < self.traceSample)) {
```

Comment:
This is incorrect.

```
if (options.hasNoParent) {
    if (Math.random() < self.traceSample) {
        options.trace = true;
    } else {
        options.trace = false;
    }
}
```

Otherwise we turn of tracing for everyone even if they are not the parent request.